tests/drivers/crc/src/main.c

```diff
@@ -268,22 +288,48 @@ ZTEST(crc, test_discontinuous_buf)
 	uint8_t data1[5] = {0x0A, 0x2B, 0x4C, 0x6D, 0x8E};
 	uint8_t data2[5] = {0x49, 0x00, 0xC4, 0x3B, 0x78};
 
-	struct crc_ctx ctx = {
+	struct crc_ctx ctx;
+
+#ifdef CONFIG_CRC_DRIVER_HAS_CRC8
+	ctx = (struct crc_ctx){
 		.type = CRC8,
 		.polynomial = CRC8_POLY,
 		.seed = CRC8_INIT_VAL,
 		.reversed = CRC_FLAG_REVERSE_INPUT | CRC_FLAG_REVERSE_OUTPUT,
 	};
+#elif defined(CONFIG_CRC_DRIVER_HAS_CRC16)
+	ctx = (struct crc_ctx){
+		.type = CRC16,
+		.polynomial = CRC16_POLY,
+		.seed = CRC16_INIT_VAL,
+		.reversed = CRC_FLAG_REVERSE_INPUT | CRC_FLAG_REVERSE_OUTPUT,
+	};
+#else
+	ctx = (struct crc_ctx){
+		.type = CRC32_C,
+		.polynomial = CRC32C_POLY,
+		.seed = CRC32_C_INIT_VAL,
+		.reversed = CRC_FLAG_REVERSE_INPUT | CRC_FLAG_REVERSE_OUTPUT,
+	};
+#endif
```

Comment:
Update as below could shorter the code changes and better for readability
```c
#ifdef CONFIG_CRC_DRIVER_HAS_CRC8

#define CRC_TEST_VARIANT CRC8
#define CRC_TEST_INIT_VAL ...
#define ...

#elif defined(CONFIG_CRC_DRIVER_HAS_CRC16)

#define CRC_TEST_VARIANT CRC16
#define CRC_TEST_INIT_VAL ...
#define ...

#else

#define CRC_TEST_VARIANT CRC32_C
#define CRC_TEST_INIT_VAL ...
#define ...

#endif
```
```suggestion
	ctx = (struct crc_ctx){
		.type = CRC_TEST_VARIANT,
		.polynomial = CRC_TEST_POLY,
		.seed = CRC_TEST_INIT_VAL,
		.reversed = CRC_TEST_REVERSE_CONFIG,
	};
```

Reply:
Please update for similar case